### ISSUE TITLE
PSR2.ControlStructures.SwitchDeclaration confused when semicolon is used for cases instead of colon

### ISSUE THREAD

**Issue by VasekPurchart** [squizlabs/PHP_CodeSniffer#579]
I got really confused by this error message:
```
 4 | ERROR | CASE statements must not be defined using curly braces
   |       | (PSR2.ControlStructures.SwitchDeclaration.WrongOpenercase)
```
In the end I narrowed it down to the problem being usage of `;` isntead of `:`. Frankly I've never known this was even possible, but it is mentioned in the [PHP manual](http://php.net/manual/en/control-structures.switch.php):

> It's possible to use a semicolon instead of a colon after a case

This can be reproduced with this code:
```php
<?php

switch ($foo) {
	case 1;
}
```

In PSR2 there is no mention if semicolon is allowed in this place so I guess the correct resolution is not to show any errors (certainly not this one).

**Comment by aik099** [comment]
I guess all switch-handling sniffs should be adapted to allow `;` usage in place of `:`.

**Comment by gsherwood** [comment]
In this case, the error message is just wrong. It is assuming there are only two ways of opening a case statement, but there are obviously three.

I'll change the message to say what you must do, and not what you must not do.

**Comment by gsherwood** [comment]
Message changed for error `PSR2.ControlStructures.SwitchDeclaration.WrongOpenercase`

Old: CASE statements must not be defined using curly braces
New: CASE statements must be defined using a colon

**Comment by VasekPurchart** [comment]
Cool, thank you :) I am glad to hear that I am not the only one who has never seen this weird format.
